openedx/core/djangoapps/course_groups/views.py

```diff
@@ -379,3 +390,356 @@ def debug_cohort_mgmt(request, course_key_string):
+            * email: Email address of the user.
+            * name: Full name of the user.
+
+    **POST Response Values**
+
+        Returns a HTTP 404 Not Found response status code when:
+            * The course corresponding to the corresponding course id could not be found.
+            * The requesting user does not have staff access to the course.
+            * The cohort corresponding to the given cohort id could not be found.
+        Returns a HTTP 200 OK response status code to indicate success.
+
+        * success: Boolean indicating if the operation was successful.
+        * added: Usernames/emails of the users that have been added to the cohort.
+        * changed: Usernames/emails of the users that have been moved to the cohort.
+        * present: Usernames/emails of the users already present in the cohort.
+        * unknown: Usernames/emails of the users with an unknown cohort.
+        * preassigned: Usernames/emails of unenrolled users that have been preassigned to the cohort.
+        * invalid: Invalid emails submitted.
+
+    Adding multiple users to a cohort, send a request to:
+    POST /api/cohorts/v1/courses/{course_id}/cohorts/{cohort_id}/users
+
+    With a payload such as:
+    {
+        "users": [username1, username2, username3...]
+    }
+
+    **DELETE Response Values**
+
+        Returns a HTTP 404 Not Found response status code when:
+            * The course corresponding to the corresponding course id could not be found.
+            * The requesting user does not have staff access to the course.
+            * The cohort corresponding to the given cohort id could not be found.
+            * The user corresponding to the given username could not be found.
+        Returns a HTTP 204 No Content response status code to indicate success.
+    """
+    serializer_class = CohortUsersAPISerializer
+
+    def get(self, request, course_key_string, cohort_id, username=None):  # pylint: disable=unused-argument
+        """
+        Lists the users in a specific cohort.
+        """
+        course_key, _ = _get_course_with_access(request, course_key_string)
+
+        try:
+            cohort = cohorts.get_cohort_by_id(course_key, cohort_id)
+        except CourseUserGroup.DoesNotExist:
+            msg = 'Cohort (ID {cohort_id}) not found for {course_key_string}'.format(
+                cohort_id=cohort_id,
+                course_key_string=course_key_string
+            )
+            raise self.api_error(status.HTTP_404_NOT_FOUND, msg, 'cohort-not-found')
+        queryset = cohort.users.all()
+        page = self.paginate_queryset(queryset)
+
+        if page is not None:
+            serializer = self.get_serializer(page, many=True)
+            return self.get_paginated_response(serializer.data)
+
+        return Response(self.get_serializer(queryset, many=True).data)
+
+    # pylint: disable=unused-argument
+    def delete(self, request, course_key_string, cohort_id, username=None):
```

Comment:
Why is the cohort_id required, if the code does not verify whether the user belonged to that particular cohort? It should either fail if the cohort_id does not match the user's cohort (for instance, the caller may have out of date information), or this endpoint should not include the cohort_id in the URL. That way, it's clear that the endpoint removes the user from their currently assigned cohort.

I'd prefer the first option, though -- verifying whether they're in the cohort passed in the url.

Reply:
The PR has been updated to validate whether the user belongs to the cohort passed in the URL.